src/main/java/me/jellysquid/mods/sodium/client/compat/CompatibilityHooks.java

```diff
@@ -0,0 +1,192 @@
+package me.jellysquid.mods.sodium.client.compat;
+
+import me.jellysquid.mods.sodium.client.SodiumClientMod;
+import me.jellysquid.mods.sodium.client.compat.impl.fabric.rendering.BlockOutlineContextImpl;
+import me.jellysquid.mods.sodium.client.compat.impl.fabric.rendering.WorldRenderContextImpl;
+import net.fabricmc.fabric.api.client.rendering.v1.InvalidateRenderStateCallback;
+import net.fabricmc.fabric.api.client.rendering.v1.WorldRenderEvents;
+import net.fabricmc.loader.api.FabricLoader;
+import net.minecraft.block.BlockState;
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.client.render.*;
+import net.minecraft.client.util.math.MatrixStack;
+import net.minecraft.client.world.ClientWorld;
+import net.minecraft.entity.Entity;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Matrix4f;
+import net.minecraft.util.profiler.Profiler;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * Utility class that initializes and holds all compatibility-related hook interfaces.
+ */
+public final class CompatibilityHooks {
+    private CompatibilityHooks() { }
+
+    public static void initialize() {
+        // This method exists purely to run the static block on startup.
+    }
+
+    private static final Logger LOGGER = SodiumClientMod.logger();
+
+    public static final FabricRenderingHooks FABRIC_RENDERING;
+
+    static {
+        final FabricLoader loader = FabricLoader.getInstance();
+        LOGGER.info("Sodium is now looking for mods to enable compatibility hooks for...");
+        FABRIC_RENDERING = createFabricRenderingHooks(loader);
+        LOGGER.info("Compatibility hooks done! Enjoy your improved performance!");
+    }
+
+    private static FabricRenderingHooks createFabricRenderingHooks(FabricLoader loader) {
```

Comment:
Since this is always called on client init, won’t it crash the game if fabric rendering api isn’t installed? Since classloading will search for the fabric api class in the return value, and not find it. Ergo, JVM crashes

Reply:
Therefore I recommend that anything that directly touches frapi be moved to a new class, which is only classloaded in the case of frapi being installed

Reply:
The code that actually references FRAPI classes is only called if FRAPI is installed (checked via `loader.isModLoaded("fabric-rendering-api-v1")`. So no, this should not result in a "class not found" error.